datalad/distribution/tests/test_publish.py

```diff
@@ -61,7 +74,8 @@ def test_publish_simple(origin, src_path, dst_path):
 
     # don't fail when doing it again
     res = publish(dataset=source, to="target")
-    eq_(res, ([source], []))
+    # and nothing is pushed
+    eq_(res, ([], []))
```

Comment:
I don't think we need two modes, but I see your point. However, we should discuss the issue of return values on a more general level. At present there is no clear concept, and the issue of what items are return is just one, the format/nature of return values (JSON-like, ...) is another. For this PR, I'd prefer to have the return values reflect what was actually done. In a second step I'd like to uniformize return values (and handling in renderers) across all or most commands.